authorize_helper_test.go

```diff
@@ -236,8 +236,11 @@ func TestIsRedirectURISecure(t *testing.T) {
 		{u: "http://localhost", err: false},
 		{u: "http://test.localhost", err: false},
 		{u: "http://127.0.0.1/", err: false},
+		{u: "http://[::1]/", err: false},
+		{u: "http://127.0.0.1:8080/", err: false},
+		{u: "http://[::1]:8080/", err: false},
 		{u: "http://testlocalhost", err: true},
-		{u: "wta://auth", err: false},
+		{u: "wta://auth", err: true},
```

Comment:
So this changes. I do not think a custom scheme is secure. On iPhone, custom app schemes are not unique to the app. I have not found a platform where this would be secure.

Reply:
> If you error here, all mobile app auth will fail in prod-systems.

Not true, they just have to use "claimed https redirects".

Or are you saying that current live production systems will start failing with this? Sure. We can have a deprecation period here. But I do think we should deprecate use of anything except for "claimed https redirects" (and loopback IPs).

Reply:
We can also offer both secure and less secure version of `IsRedirectURISecure`, where I would go with secure version by default, but changing to less secure would then be one line configuration change people have to do, to keep old behavior.

Reply:
Please revert this, as discussed in the comment above. This can not be changed, forcing our user's users to move to "claimed https redirects" because we changed the default is not acceptable.